dotnet/src/SemanticKernel.Core/Planning/InstrumentedPlan.cs

```diff
@@ -34,16 +34,29 @@ internal sealed class InstrumentedPlan : IPlan
     /// <summary>
     /// Initialize a new instance of the <see cref="InstrumentedPlan"/> class.
     /// </summary>
-    /// <param name="plan">Instance of <see cref="IPlan"/> to decorate.</param>
+    /// <param name="plan">Instance of <see cref="Plan"/> to decorate.</param>
     /// <param name="loggerFactory">The <see cref="ILoggerFactory"/> to use for logging. If null, no logging will be performed.</param>
     public InstrumentedPlan(
-        IPlan plan,
+        Plan plan,
```

Comment:
Would it be better to pass `ISKFunction plan` instead? In this case it should be easier to test `InstrumentedPlan` on our side, since we can mock `ISKFunction` (we still need to add more tests to classes with instrumentation).